src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/ReferenceSearchValueParser.cs

```diff
@@ -58,7 +57,7 @@ public ReferenceSearchValue Parse(string s)
                     return new ReferenceSearchValue(
                         ReferenceKind.InternalOrExternal,
                         null,
-                        resourceType,
+                        resourceTypeInString,
```

Comment:
We should validate to make sure this is a valid resource type (i think Enum.Parse will throw exception if it's not valid).